cookbook/design/drawer.md

```diff
@@ -0,0 +1,185 @@
+to create a layout with a Material Design Drawer!
+
+## Directions
+
+  1. Create a `Scaffold`
+  2. Add a drawer
+  3. Populate the drawer with items
+  4. Close the drawer programmatically
+  
+## 1. Create a `Scaffold`
+
+In order to add a Drawer to our app, we'll need to wrap it in a [Scaffold](https://docs.flutter.io/flutter/material/Scaffold-class.html) 
+Widget. The Scaffold Widget provides a consistent visual structure to apps that
+follow the Material Design Guidelines. It also supports special Material Design 
+components, such as Drawers, AppBars, and SnackBars.
+
+In this case, we'll want to create a `Scaffold` with a `drawer`:
+
+```dart
+new Scaffold(
+  drawer: // We'll add our Drawer here in the next step!
+);
+```
+
+## 2. Add a drawer
+
+We can now add a drawer to our `Scaffold`. A drawer could be any Widget, but 
+it's often best to use the `Drawer` widget from the [material library](https://docs.flutter.io/flutter/material/material-library.html), 
+which adheres to the Material Design spec.
+
+```dart
+new Scaffold(
+  drawer: new Drawer(
+    child: // We'll populate the Drawer in the next step!
+  )
+);
+```
+
+## 3. Populate the drawer with items
+
+Now that we have a `Drawer` in place, we can add content to it! In this example, 
+we will use a [`ListView`](https://docs.flutter.io/flutter/widgets/ListView-class.html). 
+`ListView` is handy in this situation because it will allow users to scroll 
+through the drawer if the content takes up more space than the screen supports.
+
+We will populate the `ListView` with a [`DrawerHeader`](https://docs.flutter.io/flutter/material/DrawerHeader-class.html) 
+and two [`ListTile`](https://docs.flutter.io/flutter/material/ListTile-class.html) 
+Widgets. For more information on working with Lists, please see the 
+[list recipes](/cookbook/#lists).
+
+```dart
+new Drawer(
+  // Add a ListView to the drawer. This ensures the user can scroll
+  // through the options in the Drawer if there isn't enough vertical
+  // space to fit everything.
+  child: new ListView(
+    // Important: Remove any padding from the ListView.
+    padding: EdgeInsets.zero,
+    children: <Widget>[
+      new DrawerHeader(
+        child: new Text('Drawer Header'),
+        decoration: new BoxDecoration(
+          color: Colors.blue,
+        ),
+      ),
+      new ListTile(
+        title: new Text('Item 1'),
+        onTap: () {
+          // Update the state of the app
+          // ...
+        },
+      ),
+      new ListTile(
+        title: new Text('Item 2'),
+        onTap: () {
+          // Update the state of the app
+          // ...
+        },
+      ),
+    ],
+  ),
+);
+```
+
+## 4. Close the drawer programmatically
+
+After a user taps on an item, we often want to close the drawer. How can we 
+achieve this? Using the [Navigator](https://docs.flutter.io/flutter/widgets/Navigator-class.html)!
+When a user taps on an item, we can call `Navigator.pop(context)` to close the
+drawer.
```

Comment:
Explain? "When user opens the Drawer, what they're implicitly doing is navigating to it. The Drawer is put on top of the Navigator stack, so to get back to whatever the user was doing before opening the Drawer, we pop the stack." 

Or something.